fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/domain/SinglePrefer.java

```diff
@@ -41,7 +41,13 @@
      * @throws ParseException if parse exception occurred
      */
     public SinglePrefer(final String header) throws ParseException {
-        preferTags.addAll(HttpHeaderReader.readList(PreferTag::new, header));
+        preferTags.addAll(stream(header.split(",")).map(h -> {
+            try {
+                return new PreferTag(h);
+            } catch (final ParseException e) {
+                throw new IllegalArgumentException("Could not parse 'Prefer' header", e);
+            }
+        }).collect(toSet()));
```

Comment:
Mos def. The house style has evolved to minimize checked exceptions and when they are used, to give them very specific semantics (e.g. not `RepositoryException` but `OutOfDomainSubjectException`).